allneeds.lua

```diff
@@ -0,0 +1,45 @@
+-- Prints the sum of all citizens' needs.
+
+local fort_needs = {}
+for _, unit in pairs(df.global.world.units.all) do
+    if not dfhack.units.isCitizen(unit) or not dfhack.units.isAlive(unit) then
+        goto skipunit
+    end
+
+    local mind = unit.status.current_soul.personality.needs
+    -- sum need_level and focus_level for each need
+    for _,need in pairs(mind) do
+        if fort_needs[need.id] then
+            fort_needs[need.id].cummulitative_need = fort_needs[need.id].cummulitative_need + need.need_level
+            fort_needs[need.id].cummulitative_focus = fort_needs[need.id].cummulitative_focus + need.focus_level
+            fort_needs[need.id].citizen_count = fort_needs[need.id].citizen_count + 1
+        else
+            fort_needs[need.id] = {}
+            fort_needs[need.id].cummulitative_need = need.need_level
+            fort_needs[need.id].cummulitative_focus = need.focus_level
+            fort_needs[need.id].citizen_count = 1
+        end
```

Comment:
```suggestion
        local needs = ensure_key(fort_needs, need.id)
        needs.cumulative_need = (needs.cumulative_need or 0) + need.need_level
        needs.cumulative_focus = (needs.cumulative_focus or 0) + need.focus_level
        needs.citizen_count = (needs.citizen_count or 0) + 1
```

Reply:
this would shorten up the logic in this section

Reply:
`ensure_key` is neat, thanks :)